pkg/kube/kube_handler.go

```diff
@@ -278,6 +287,24 @@ func (a *userHandler) DeleteAppRepository(repoName, repoNamespace string) error
 	return err
 }
 
+func (a *userHandler) ValidateAppRepository(appRepoBody io.ReadCloser) (*http.Response, error) {
```

Comment:
A similar setup here would allow testing the URL path trimming - not sure it's necessary though if the handler above is tested.